tests/core/engine_adapter/test_risingwave.py

```diff
@@ -15,6 +15,43 @@ def adapter(make_mocked_engine_adapter):
     return adapter
 
 
+def test_columns(adapter: t.Callable):
```

Comment:
Let's include an integration test for this please, containing the problematic case you described in the description. You can create a `test_integration_risingwave`, similar to other integration testing modules.